cmake/kokkos_enable_options.cmake

```diff
@@ -67,6 +67,7 @@ mark_as_advanced(Kokkos_ENABLE_IMPL_MDSPAN)
 mark_as_advanced(Kokkos_ENABLE_MDSPAN_EXTERNAL)
 mark_as_advanced(Kokkos_ENABLE_IMPL_SKIP_COMPILER_MDSPAN)
 
+KOKKOS_ENABLE_OPTION(CUDA_LAMBDA ON "Whether to allow lambda expressions on the device with NVCC **DEPRECATED**")
```

Comment:
The docstring was updated as compared to 4.0 that had `"Whether to activate experimental lambda features"`.